Written about some of the prototypes (#20)
SW705E19/report: sections/design/prototypes.tex

\section{Prototypes}
To get a clear view of how the different pages will look and catch any early potential design problems, we have created prototypes for each of the different pages. 

> We will use these to make the implementation of the different pages easier make ensure that we have a consistent design througout the system. 

Reviewer: ```suggestion
We will use these to make the implementation of the different pages easier to ensure that we have a consistent design througout the system. 
```

The prototypes will not serve as an exact vision of how the pages will look but will instead serve as a guideline to how the page should be set up.
In this section we will look at some of the more interesting prototypes where we have had to make some design choices.
\begin{figure}[H]
    \includegraphics[width=\linewidth]{/prototypes/admin-dashboard.png}
     \caption{Shows the admin dashboard for admins that are logged into the system}
     \label{fig:admin-dashboard}
 \end{figure}
The admin dashboard gives the admin an overview of the recently reported tutors and the different categories. 

> The admin needs to see tutor reports since they are the once that will handle them and decide if they should be banned. 

Reviewer: ```suggestion
The admin needs to see tutor reports since they are the ones that will handle them and decide if a tutor should be banned. 
```

> Admins are also in charge of the different teaching categories and therefore they need to have an overview of the current ones and also be able to add new.

Reviewer: ```suggestion
Admins are also in charge of the different teaching categories and therefore they need to have an overview of the current ones and also be able to add new categories.
```

> We also have a top bar that is used on every page for when the user is logged in. It shows a couple of different icons where the user can go to the inbox, the calendar or the user's own page. It will also show notifications for when different events related to the user happens in the system. On the far left side of the top bar we consider having an icon to go to the landing page but this have not been decided yet

Reviewer: ```suggestion
We also have a top bar that is used on every page for when the user is logged in. It shows a couple of different icons where the user can go to the inbox, the calendar or the user's own page. 
It will also show notifications for when different events related to the user happen in the system. 
On the far left side of the top bar we considered having an icon to go to the landing page but this has not been decided yet
```

Reviewer: ift sidste sætning - hvad så når det bliver decided, skal vi opdatere? tror vi skal tage en executive decision før vi merger det her ind og så enten slette eller skrive at den er der

\begin{figure}[H]
   \includegraphics[width=\linewidth, scale=0.5]{/prototypes/landing-page.png}
    \caption{The landing page for the system}
    \label{fig:landing-page}
\end{figure}
The landing page is the first page that the users of the system will see, and thus it is important that this page shows some interesting content. 
This page can be accessed both with and without being logged in. 

> We have a top bar that change depending on if the user is logged in or not. 

Reviewer: ```suggestion
We have a top bar that changes depending on if the user is logged in or not. 
```

> In the top we will have a carousel that shows some hype images.

Reviewer: ```suggestion
In the top of the landing page we will have a carousel that shows some interesting images.
```

Underneath we have a search functionality to find a course or a specific tutor. 
The user should be able to search for both tutors and courses. 

> Finally there will be some recommendations for the user which will suggest specific tutors and also show the most popular courses. 

Reviewer: ```suggestion
Finally, there will be some recommendations for the user which will suggest specific tutors and also show the most popular courses. 
```

 \begin{figure}[H]
    \includegraphics[width=\linewidth]{/prototypes/inbox.png}
     \caption{The inbox for the messaging component in the system}
     \label{fig:inbox}
 \end{figure}
When the user clicks the message icon in the top bar they will be redirected to the inbox page. Here they can view their recent messages. 

> We might also include a search functionality to find specific messages. 

Reviewer: ```suggestion
We considered including a search functionality to find specific messages, but ultimately did not add it to the prototypes. 
```

The most recently received messages should be shown on the top of the list so they are easy to find. 
When a message is received the topbar should also show a notification for this. 
There should also be something to indicate that a message is unread. 
The user can click a button on the left side of the page to start writing a new message. 

 \begin{figure}[H]
    \includegraphics[width=\linewidth]{/prototypes/make-appointment.png}
     \caption{Page that shows a tutor's calendar where the student can request an appointment}
     \label{fig:make-appointment}
 \end{figure}
 This page is the tutor's calendar from a student's point of view. 
 The student is able to browse the tutor's calendar and look for a date where the tutor is available, shown as the color green. 
 The student then clicks the desired date and the popup window will show. 

>  Here they can specify the subject they wish to be tought and the timeslot as well as additional information. 

Reviewer: ```suggestion
 Here they can specify the subject they wish to be taught and the timeslot as well as additional information. 
```

 They can only choose one of tutors available subjects, and the timeslot should comply with the tutor's availability. 
 The tutor also needs to choose what time they are available prior to this.
 When the student makes the appointment they will be sent to a payment page, and a request will be sent to the tutor which they can either accept or deny. 

 \begin{figure}[H]
    \includegraphics[width=\linewidth]{/prototypes/view-information-on-tutor.png}
     \caption{The page where other users can view information about a specific tutor}
     \label{fig:view-information-on-tutor}
 \end{figure}
This is the page where other users can go and view information about another tutor. 
A lot of information should be available here so that a user can know if this tutor can be helpful to them.

> The page will show basic information on the left side and also show the tutors rating. 

Reviewer: ```suggestion
The page will show basic information on the left side and also show the tutor's rating. 
```

A user can click \textit{see reviews} to see what other users wrote about the tutor. 
The user viewing the tutor's page can also leave a review of the tutor or report the tutor if they had a bad experience. 

> The tutor can display some self written information about them in the middle that others can read. The middle also shows which subjects that the tutor teaches. 

Reviewer: ```suggestion
The tutor can display some written information about them in the middle that others can read. 
The middle also shows which subjects the tutor teaches. 
```

A tutor also has some material connected to their page if they have uploaded any. 
They can make this available to others however they wish, but can also lock it behind payment or only make it available if a student books an appointment with the tutor.
Users can click the \textit{View all material} button to go to the view material page where all the files and links can be found. 
Finally the user can wish to make an appointment with the tutor. 

> By clicking the \textit{Make appointment} button the user is redirected to the tutors calendar where they can find a suitable date for the appointment. 

Reviewer: ```suggestion
By clicking the \textit{Make appointment} button the user is redirected to the tutor's calendar where they can find a suitable date for the appointment. 
```

 \begin{figure}[H]
    \includegraphics[width=\linewidth]{/prototypes/view-tutor-material.png}
     \caption{The page that shows a tutors available material}
     \label{fig:view-tutor-material}
 \end{figure}

> This page shows a specific tutors uploaded material. The material is divided into which subject the it belongs to make it easier for other users to find the relevant material. The material can be any kind of file or it can be links to useful pages such as some reading material or perhaps a youtube video. When a specific user goes to a tutor's material page, only the material that is accessible to them should be made available. All other files should show a lock on top of the file or link to indicate that this item is not available. The user can then hover over this item and it should then tell the user how it can be unlocked. 

Reviewer: ```suggestion
This page shows a specific tutor's uploaded material. 
The material is divided into which subject it belongs to, in order to make it easier for other users to find the relevant material. 
The material can be any kind of file or it can be links to useful pages such as some reading material or perhaps a youtube video. 
When a specific user goes to a tutor's material page, only the material that is accessible to them should be made available. 
All other files should show a lock on top of the file or link to indicate that this item is not available. 
The user can then hover over this item and it should then tell the user how it can be unlocked. 
```
